### ISSUE TITLE
loosing all tasks when migrating to new persistence store

### ISSUE THREAD

**Issue by vixns** [mesosphere/marathon#4400]
While upgrading to latest marathon, the zookeeper store was reimplemented, the new tree is fine, but during the migration step, I've loose all my apps and groups, then all existing task were killed.
 
It may be related to non-english locale (french in this case)

```
[WARN] [09/15/2016 20:25:40.734] [marathon-akka.actor.default-dispatcher-33] [akka://marathon/user/IO-HTTP/group-0/24] Illegal response header: Illegal 'Date' header: Invalid input 'j', expected $timesDate (line 1, pos 1):
jeu., 15 sept. 2016 20:25:40 GMT
^
```

Then, after the migration and marathon restart
```
WARN App /test exists in TaskTracker, but not App store. The app was likely terminated. Will now expunge. (mesosphere.marathon.SchedulerActions:ForkJoinPool-2-worker-3)
```

( note to myself and the universe: always get /v2/groups before upgrading, it will save your day )

**Comment by jasongilanfarr** [comment]
Can you send me the groups json? We stored all dates in ISO8601 in both versions, so this seems off to me.

**Comment by vixns** [comment]
All dates are in ISO8601 in the json and zookeeper.
This date is the operation date,`[09/15/2016 20:25:40.734]` at the beginning of the log line.

**Comment by jasongilanfarr** [comment]
Can you include the lines around migration itself? You aren't supposed to lose anything while migrating to the new format.

**Comment by vixns** [comment]
I've inspected more logs, the date warning appends from a long time, it's not related to this issue.

I don't see anything else on the logs, maybe because of the "warn"  logging level.
I have another migration to do in a few hours, I will enable debug first and keep all logs.
Any advice to generate useful traces ?

**Comment by jasongilanfarr** [comment]
There's a line from migration that says how much it migrated. That line would be very helpful. I'll try to reproduce locally as well.

**Comment by jasongilanfarr** [comment]
I know that I fixed a major bug in migration in the last week (it had these symptoms- so it may already be fixed)- funny enough, if you reverted your root, it'd magically show up. Could be wrong, but prior to my recent fixes, there was a major issue in migration.

**Comment by jasongilanfarr** [comment]
I believe this is fixed via #4322 - there's one more blocker due to a race condition. I'm hoping to fix that today. Sometimes deployment plans will fail as the root wasn't stored first. I think it's relatively harmless but annoying none the less.

**Comment by vixns** [comment]
I've made another upgrade, with a fresh marathon build from HEAD.

After the migration, only 3 of 59 apps remains.

The interesting log parts:

```
Sep 17 00:16:16 marathon[22026]: [DEBUG] [09/17/2016 00:16:16.941] [marathon-akka.actor.default-dispatcher-23] [akka://marathon/user/taskKillServiceActor/1] Adding 1 tasks to queue; setting up child actor to track progress
Sep 17 00:16:16 marathon[22026]: [INFO] [09/17/2016 00:16:16.944] [marathon-akka.actor.default-dispatcher-13] [akka://marathon/user/taskKillServiceActor/1/$a] Starting TaskKillProgressActor686892934 to track kill progress of 1 tasks
Sep 17 00:16:16 marathon[22026]: [INFO] [09/17/2016 00:16:16.944] [marathon-akka.actor.default-dispatcher-23] [akka://marathon/user/taskKillServiceActor/1] processing 1 kills
Sep 17 00:16:16 marathon[22026]: [INFO] [09/17/2016 00:16:16.950] [marathon-akka.actor.default-dispatcher-4] [akka://marathon/user/MarathonScheduler/$a] Deployment of / successful
Sep 17 00:16:16 marathon[22026]: [INFO] [09/17/2016 00:16:16.950] [marathon-akka.actor.default-dispatcher-17] [akka://marathon/user/MarathonScheduler/$a/DeploymentManager] Removing 03817701-7f15-4e65-87bb-ca51d064f0f5 from list of running deployments
Sep 17 00:16:16 marathon[22026]: [2016-09-17 00:16:16,966] ERROR While retrieving 03817701-7f15-4e65-87bb-ca51d064f0f5, either original (Some(Group(/,Map(),Set(),Set(),2016-09-16T22:16:14.598Z))) or target (None) were no longer available (mesosphere.marathon.storage.repository.StoredPlan:ForkJoinPool-2-worker-95)
```

```
2016-09-16T22:16:26.84038 : Sep 17 00:16:26 marathon[22026]: [2016-09-17 00:16:26,840] ERROR While retrieving bbec174f-d83e-4fb5-8a3b-2048cd7f0dd6, either original (Some(Group(/,Map(),Set(Group(/test,Map(),Set(Group(/test/test1,Map(),Set(Group(/test/test1/web,Map(/test/test1/web/mysql -> AppDefinition(/test/test1/web/mysql,None,None,None,Map(MYSQL_ROOT_PASSWORD -> EnvVarString(***************), MYSQL_PASSWORD -> EnvVarString(***************), MYSQL_USER -> EnvVarString(yuki), MYSQL_DATABASE -> EnvVarString(yuki)),1,0.02,400.0,0.0,0,,Set(),Vector(),Vector(),Vector(),false,1000 milliseconds,1.15,3600000 milliseconds,Some(MesosDocker(Vector(DockerVolume(/var/lib/mysql,/srv/data/pp/test1/web/mysql,RW)),mysql/mysql-server,None,false)),Set(MarathonTcpHealthCheck(300 seconds,60 seconds,20 seconds,3,None,Some(3306))),Vector(),None,Set(),UpgradeStrategy(0.0,1.0),Map(),None,Some(IpAddress(Vector(),Map(),DiscoveryInfo(List(Port(3306,mysql,tcp,M...
2016-09-16T22:16:26.84049 : Sep 17 00:16:26 marathon[22026]: (2016-09-16T22:16:16.665Z,2016-09-16T22:16:16.665Z,2016-09-16T22:16:16.665Z),None,Map())),Set(),Set(),2016-09-16T22:16:23.725Z)),Set(),2016-09-16T22:16:23.725Z), Group(/test/test2,Map(/test/test2/mysql -> AppDefinition(/test/test2/mysql,None,None,None,Map(MYSQL_ROOT_PASSWORD -> EnvVarString(***************), MYSQL_PASSWORD -> EnvVarString(***************), MYSQL_USER -> EnvVarString(yuki), MYSQL_DATABASE -> EnvVarString(yuki)),1,0.02,400.0,0.0,0,,Set(),Vector(),Vector(),Vector(),false,1000 milliseconds,1.15,3600000 milliseconds,Some(MesosDocker(Vector(DockerVolume(/var/lib/mysql,/srv/data/pp/test2/mysql,RW)),mysql/mysql-server,None,false)),Set(MarathonTcpHealthCheck(300 seconds,60 seconds,20 seconds,3,None,Some(3306))),Vector(),None,Set(),UpgradeStrategy(0.0,1.0),Map(),None,Some(IpAddress(Vector(),Map(),DiscoveryInfo(List(Port(3306,mysql,tcp,Map()))),Some(cnilan))),FullVersionInfo(2016-09-16T22:16:23.725Z,201...
2016-09-16T22:16:26.84057 : Sep 17 00:16:26 marathon[22026]: 16:23.725Z),None,Map())),Set(),Set(),2016-09-16T22:16:23.725Z)),Set(),2016-09-16T22:16:23.725Z)),Set(),2016-09-16T22:16:23.725Z))) or target (None) were no longer available (mesosphere.marathon.storage.repository.StoredPlan:ForkJoinPool-2-worker-77)
```

```
Sep 17 00:16:30 marathon[22026]: [INFO] [09/17/2016 00:16:30.458] [marathon-akka.actor.default-dispatcher-7] [akka://marathon/user/MarathonScheduler/$a] initiate task reconciliation
Sep 17 00:16:30 marathon[22026]: [2016-09-17 00:16:30,462] WARN App /test/test1/app exists in TaskTracker, but not App store. The app was likely terminated. Will now expunge. (mesosphere.marathon.SchedulerActions:ForkJoinPool-2-worker-127)
```

marathon launch options
```
INFO Starting Marathon 1.4.0-SNAPSHOT/0a0ec705be42e44246eb116d82aa1a27dc5d3194 with --plugin_dir /var/lib/marathon/plugins --leader_proxy_ssl_ignore_hostname --ssl_keystore_password ****** --plugin_conf /etc/marathon/plugin.conf --master zk://10.36.0.1:2181/mesos --logging_level warn --mesos_role marathon --max_tasks_per_offer 10 --enable_features task_killing,external_volumes --mesos_authentication_principal marathon --mesos_authentication --http_realm "Marathon" --mesos_authentication_secret_file /etc/marathon/secret --disable_http --zk zk://10.36.0.1:2181/marathon --ssl_keystore_path /etc/marathon/marathon.jks --hostname 10.36.0.1 (mesosphere.marathon.Main$:main)
```

**Comment by jasongilanfarr** [comment]
@vixns Thanks for investigating further, the latter error you hit was fixed on Friday with #4480. I just deployed a local marathon and verified everything works as expected. Please try master after aed79ff79dd6b17153a0559069842f8e09017d75 and let me know if you run into anything further.
